site/app/libraries/FileUtils.php

```diff
@@ -501,13 +538,25 @@ public static function validateUploadedFiles($files) {
         for ($i = 0; $i < $num_files; $i++) {
             //extract the values from each file
             $name = $files['name'][$i];
-            $type = mime_content_type($files['tmp_name'][$i]);
-            $size = $files['size'][$i];
-            $err_msg = "";
-
-            //did anything go wrong?
+            $tmp_name = $files['tmp_name'][$i];
+            $type = mime_content_type($tmp_name);
+            
+            $zip_status = FileUtils::checkZipFileStatus($tmp_name);
```

Comment:
can `\ZipArchive` handle anything beyond `application/zip` content-types (which we get from the `$type` variable right above this)? I don't believe so, and so we don't really need to waste time trying to do anything with zips if `$type !== 'application/zip'`.

I'd probably rewrite things as:
```
$is_zip = $type === 'application/zip';
if ($is_zip) {
    $zip_status = FileUtils::getZipFileStatus($tmp_name);
    if ($zip_status !== \ZipArchive::ER_OK) {
        $err_msg = ErrorMessages::getZipErrorMessage($zip_status);
    }
    else {
        $size = FileUtils::getZipSize($tmp_name);
        if (!FileUtils::checkFileInZipName($tmp_name)) {
            $err_msg = "Invalid filename within zip file"; 
        }
    }
}
```